internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -104,6 +117,7 @@ type xr struct {
 	managed.ConnectionDetailsFetcher
 	ComposedResourceObserver
 	ComposedResourceGarbageCollector
+	ExtraResourcesGetter
```

Comment:
Nit: Would `Fetcher` or `Observer` work here, to better match the two other interfaces that "get stuff"?